app/src/main/java/de/christinecoenen/code/zapp/app/player/CustomTrackNameProvider.java

```diff
@@ -0,0 +1,140 @@
+package de.christinecoenen.code.zapp.app.player;
+
+import android.content.res.Resources;
+import android.text.TextUtils;
+import com.google.android.exoplayer2.C;
+import com.google.android.exoplayer2.Format;
+import com.google.android.exoplayer2.ui.DefaultTrackNameProvider;
+import com.google.android.exoplayer2.util.MimeTypes;
+import com.google.android.exoplayer2.util.Util;
+import de.christinecoenen.code.zapp.R;
+
+import java.util.Locale;
+
+
+public class CustomTrackNameProvider extends DefaultTrackNameProvider {
+
+	private final Resources resources;
+	private Format tempFormat = null;
+
+	/**
+	 * @param resources Resources from which to obtain strings.
+	 */
+	CustomTrackNameProvider(Resources resources) {
+		super(resources);
+		this.resources = resources;
+	}
+
+	@Override
+	public String getTrackName(Format format) {
+		String trackName;
```

Comment:
This class can be heavily simplified by making a call to `super.getTrackName(fomat)`. The resulting string can then be suffixed if needed. All other methods than can be deleted.

Reply:
You are totally right. This class could be more simple. I will refractor it tomorrow and will push my changes. Thanks a lot for your feedback.